widgetssdk/src/main/java/com/glia/widgets/GliaWidgets.java

```diff
@@ -195,17 +196,42 @@ public static void endEngagement() {
      * Updates the visitor's information
      * <p>
      * Updates the visitor's information stored on the server. This information will also be displayed to the operator.
+     *
+     * @deprecated since 1.9.0 use @see {@link #pushVisitorInfo(VisitorInfoUpdateRequest, Consumer)}
      */
+    @Deprecated
     public static void updateVisitorInfo(VisitorInfoUpdate visitorInfoUpdate, Consumer<GliaWidgetException> exceptionConsumer) {
-        Dependencies.glia().updateVisitorInfo(new VisitorInfoUpdateRequest.Builder()
+        pushVisitorInfo(new VisitorInfoUpdateRequest.Builder()
                 .setName(visitorInfoUpdate.getName())
                 .setEmail(visitorInfoUpdate.getEmail())
                 .setPhone(visitorInfoUpdate.getPhone())
                 .setNote(visitorInfoUpdate.getNote())
                 .setCustomAttributes(visitorInfoUpdate.getCustomAttributes())
                 .setCustomAttrsUpdateMethod(visitorInfoUpdate.getCustomAttrsUpdateMethod())
                 .setNoteUpdateMethod(visitorInfoUpdate.getNoteUpdateMethod())
-                .build(), e -> {
+                .build(), exceptionConsumer);
+    }
+
+    /**
+     * Updates the visitor's information
+     * <p>
+     * Updates the visitor's information stored on the server. This information will also be displayed to the operator.
+     * <pre>
+     *     <code>
+     *         Map  customAttributes = new HashMap<>();
+     *              customAttributes.put("age", "21+");
+     *              customAttributes.put("custom_id", "H5AC76");
+     *
+     *         VisitorInfoUpdateRequest visitorInfoUpdate = new VisitorInfoUpdateRequest.Builder()
+     *                       .setName("John Doe")
+     *                       .setEmail("example@email.com")
+     *                       .setCustomAttributes(customAttributes)
+     *                       .build()
+     *     </code>
+     * </pre>
+     */
+    public static void pushVisitorInfo(VisitorInfoUpdateRequest visitorInfoUpdate, Consumer<GliaWidgetException> exceptionConsumer) {
```

Comment:
Do we need to have this interface here? Could integrator use Core SDK interface for updating visitor info directly?

Reply:
As far as I know, the integrator uses `GliaWidgets` interface to deal with the Core SDK(`Glia`) interface, that's why I think at the moment it would be better to deal with one only interface to deal with our SDK. I think we need to discuss these changes with the android team.